---
title: "Meeting Notes"
output: 
  html_notebook: 
    highlight: tango
    theme: cerulean
    toc: yes
---
# March 6th, 2018 (kat)

We should use python to standardize our fasta format... what I mean by fasta format is the information contained in the header. Some programs will require a specific set of information in the header, honestly I dont remember exactly what it is but using python it shouldnt be hard to deal with that.

Work flow so far:

  1. Extract Fasta files (SRA: https://trace.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?view=toolkit_doc)

  2. Convert the header format (Python, we should use a line by line format so we use as little computational power as possible, which is how I did my assignment for the course so we would have a starting point)

  3. Standardize the quality of our sequences... this is a simple piece of code but quite complicated theoretically. What choices do we make? Do we choose the standard? Should we be more difficult since were looking at a lot of sources? **We should discuss this** 
  
Please become familiar with TRIMMOMATIC: 

  website: http://www.usadellab.org/cms/?page=trimmomatic
  
  manual: http://www.usadellab.org/cms/uploads/supplementary/Trimmomatic/TrimmomaticManual_V0.32.pdf

  4. QIIME to assign OTU (https://qiime2.org/)... PROBLEM: QIIME is on the cluster not QIIME2... what should we do?
  
  5. PICRUst (http://picrust.github.io/picrust/)
  
  
# March 2nd, 2018 (kat)

Jeff says "I was looking at EBI Metagenomics / ENA European Nucleotide Archive"

+ Good point: we will need to make sure that the samples we take dont over lap.

**CHANGE IN Keywords: endophyte, 16s**

Then we search for illumina.

---------------------------------------------------------------------------------

# Feb 22, 2018 (by Hanna)

We all have a database

Kat: pick from list of repositories (https://www.nature.com/sdata/policies/repositories)

Jeff: pick from list of repositories (https://www.nature.com/sdata/policies/repositories)

Brody: Dryad

Hannah: SRA 




  
**Keywords: endophyte, plants, illumina, 16s**
 



Download fasta files and put them on the cluster:

* Every database has a different format

* It will be confusing

* Have to figure out what theyâ€™re giving you

    i.e. each individual or a scramble in each fasta file

    do they give phred score? 

* Better to get quality score if possible, but we can calculate
 



Information to find for each sample **(Could try to contact authors)**: 

* location of species

* Treatment

* Year of experiment

* Illumina platform

* Linked paper if possible

  We could include the type of barcoding in our analysis (could bias sequence and abundance of illumina sequencing).
 
16s:

* Usually people look for v4

* Take note of which area of the 16s they looked at

* Also which **specific barcode they used**
 
*Goal: ten samples each max*
 

 
Notes about the flow to expect after data mining:

  Select data

  Quality filtering

  Qiime 2 and OTU assignment

  Functional assignment in picrust (KEGG orthology database)
  
  Statistical analysis (mostly multivariate... I recommend you get familiar with Numerical Ecology in R)
