---
title: "Meeting Notes"
output:
  html_document:
    df_print: paged
    toc: yes
  html_notebook:
    highlight: tango
    theme: cerulean
    toc: yes
---
# April 9th, 2018 (Hannah)

- Fixed trimmomatic script
- Fixed SRAtoILLUMINA.py script
- Ran trimmomatic_job - some errors *see Trimmomatic_3.err*
- Wrote Gunzip.sh script

We need to identify trimmomatic errors, then run Gunzip.sh to decompress all files, and then run QIIME

Note: I decompressed Pylaisiella files in the command line to check the script (so they are unzipped)


# April 6th, 2018 (Brody)

Wrote and ran script to count number of fastq files in each data folder - posted script to Github

Continuously writing meta_data.csv file, here are the columns so far:

project_id | folder | found_by | number_samples | host_species | lat | long | location | date | source | data_link | paper_link | barcodes | lab_or_field_collection


# April 5th, 2018 (Brody)

Updated "data sources" spreadsheet with correct species for Pinus_flexilis - this dataset includes both Pinus flexilis and Picea engelmanni


# March 27th, 2018 (Brody)

Added folder in group endophyte project folder titled "Quercus" which contains fasta files from the dataset I retrieved from DRYAD.



# March 26th, 2018 (Kat)

Katherine cluster info:

ssh sa104007@login.cac.queensu.ca
Leo1618Wednesday


+ PREVIOUS PROJECT READS

 - PRJNA288893: https://www.sciencedirect.com/science/article/pii/S2213596015300313
 
 Aloe vera (Aloe)
 
 4 samples
 
 Illumina MiSeq
 
 Malaysia: Selangor,Sungai Buloh
 
 
 -PRJNA354831:https://www.ncbi.nlm.nih.gov/Traces/study/?acc=SRP093862
 
 Zea mays (maize)
 
 39 N 116 E
 
 Illumina HiSeq 2500
 
 4 (2 high p/ 2 low p)
 
 
 -PRJNA360379:https://www.sciencedirect.com/science/article/pii/S2213596017300314
 
 Oryza sativa (rice) 
 
 22.57 N 88.36 E
 
 Illumina HiSeq 1000
 
 2 samples


 -PRJNA343393:https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5628320/
 
 Paeonia suffruticosa (tree peony) 
 
 Illumina MiSeq
 
 China:Luoyang National Peony Garden
 
 

# March 22nd, 2018 (Brody)

### Poster titles

If we don't like any of these, please suggest another title!

"A meta-analysis of the functional diversity of endophytes"

**"Does host plant species predict associations with functional endophyte communities?"** winner!

"Do plant species associate with endophyte communities according to function?"

"Plant hosts and the specificity of endophyte functional groups: A meta-analysis"




# March 21st, 2018 (Katherine)

Project Numbers found in the last third of the SRA database search:

- PRJNA343586: https://www.ncbi.nlm.nih.gov/Traces/study/?acc=SRP091023
  Chladophora oligoclona endophytes. 
  10 samples. 
  Submitted by Capital Normal University in Beijing, China. 
  Illumina MiSeq.
  *have barcodes.
  
- PRJNA325438: https://www.ncbi.nlm.nih.gov/Traces/study/?acc=SRP078033

  Orchid Phalenopsis
  
  1 sample
  
  Submitted by Kerala Agricultural University in Thrissur, India. 
  
  Illumina MiSeq 
  
  locus tag: BAU12
  
  *Don't seem to have barcodes.

- PRJNA307272: https://www.ncbi.nlm.nih.gov/Traces/study/?acc=SRP067933

  Pinus flexilis
  
  7 samples
  
  Submitted by University of California in Merced, USA
  
  Illumina MiSeq
  
  *Don't seem to have barcodes.
 
- PRJEB7842: https://www.ncbi.nlm.nih.gov/Traces/study/?acc=ERP008816

  Populus balsamifera
  
  150 samples
  
  Canada	lat:50.52	long:-103.64
  
  Illumina MiSeq
  
  *Title could be decyphered with work
  
- PRJEB7137:https://www.ncbi.nlm.nih.gov/Traces/study/?acc=PRJEB7137&go=go

  Phelipanche aegyptiaca (parasitic plant) AND Solanum lycopersicum (host)
  
  9 samples (5/4)
  
  Submitted by Newe Ya'ar Agricultural Research Organization in Israel
  
  Illumina MiSeq
  
  *have barcodes.
  
- PRJNA288708 :https://www.ncbi.nlm.nih.gov/Traces/study/?acc=SRP060388

  Espeletia
  
  2 samples (Young and Old) + 2 samples total bacteria...
  
  El Coquito, Santa Marta, Magdalena, Colombia
  
  Illumina MiSeq
  
  *Don't seem to have barcodes.
  
Raw list:

- PRJNA343586
- PRJNA325438
- PRJNA307272
- PRJEB7842
- PRJEB7137
- PRJNA288708


# March 20th, 2018 (Hannah)

Project Numbers:

- PRJNA381586 
  - Brassica napus (Impact of dibutyl phthalate (DBP) on properties and bacteria community of soil and oil rape)
  
- PRJNA310879
  - Brassica oleracea
  
- PRJNA327111
  - Pinus pinaster (analyzed the effects of the invasion of Bursaphelenchus xylophilus, the major pathogen causing pine wilt disease (PWD), on the endophytic microbiome of adult P. pinaster trees
  
- PRJNA327111 **NOTE BY KAT: Why do you have a repeat?**


# March 20th, 2018 (Hannah)

```
#!/bin/bash
#SBATCH -c 12
#SBATCH --time=1-10:00:00
#SBATCH --job-name=SRA_2
#SBATCH --mail-type=ALL
#SBATCH --mail-user=h_driver23@hotmail.com
#SBATCH --error=SRA_2.err

cd "/global/home/sa104006/fastq/SRA2/"
for ((i = 32; i<= 55; i++)); do "/global/home/sa104006/SRA/sratoolkit.2.9.0-ubuntu64/bin/fastq-dump.2.9.0" --accession SRX25019$i$; done


```



# March 14th, 2018 (Brody)

Brody's keyword searches:

"search term" - number of relevant hits

"endophyte amplicon illumina" - 0

"endophyte amplicon" - 1

"endophyte illumina" - 1

"endophyte 16s" - 1


# March 7th, 2018 (kat)

**CHANGE IN Keywords: endophyte, AMPLICON, illumina**

https://www.ncbi.nlm.nih.gov/sra/?term=endophyte+AND+amplicon+AND+illumina --> Hannah/ Jeff/ Kat get 1/3 of the results each...

Brody stays with DRYAD.


# March 6th, 2018 (kat)

We should use python to standardize our fasta format... what I mean by fasta format is the information contained in the header. Some programs will require a specific set of information in the header, honestly I dont remember exactly what it is but using python it shouldnt be hard to deal with that.

Work flow so far:

  1. Extract Fasta files (SRA: https://trace.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?view=toolkit_doc)

  2. Convert the header format (Python, we should use a line by line format so we use as little computational power as possible, which is how I did my assignment for the course so we would have a starting point)

  3. Standardize the quality of our sequences... this is a simple piece of code but quite complicated theoretically. What choices do we make? Do we choose the standard? Should we be more difficult since were looking at a lot of sources? **We should discuss this** 
  
Please become familiar with TRIMMOMATIC: 

  website: http://www.usadellab.org/cms/?page=trimmomatic
  
  manual: http://www.usadellab.org/cms/uploads/supplementary/Trimmomatic/TrimmomaticManual_V0.32.pdf

  4. QIIME to assign OTU (https://qiime2.org/)... PROBLEM: QIIME is on the cluster not QIIME2... what should we do?
  
  5. PICRUst (http://picrust.github.io/picrust/)
  
  
# March 2nd, 2018 (kat)

Jeff says "I was looking at EBI Metagenomics / ENA European Nucleotide Archive"

+ Good point: we will need to make sure that the samples we take dont over lap.

**CHANGE IN Keywords: endophyte, 16s**

Then we search for illumina.

---------------------------------------------------------------------------------

# Feb 22, 2018 (by Hannah)

We all have a database

Kat: pick from list of repositories (https://www.nature.com/sdata/policies/repositories)

Jeff: pick from list of repositories (https://www.nature.com/sdata/policies/repositories)

Brody: Dryad

Hannah: SRA 




  
**Keywords: endophyte, plants, illumina, 16s**
 



Download fasta files and put them on the cluster:

* Every database has a different format

* It will be confusing

* Have to figure out what theyâ€™re giving you

    i.e. each individual or a scramble in each fasta file

    do they give phred score? 

* Better to get quality score if possible, but we can calculate
 



Information to find for each sample **(Could try to contact authors)**: 

* location of species

* Treatment

* Year of experiment

* Illumina platform

* Linked paper if possible

  We could include the type of barcoding in our analysis (could bias sequence and abundance of illumina sequencing).
 
16s:

* Usually people look for v4

* Take note of which area of the 16s they looked at

* Also which **specific barcode they used**
 
*Goal: ten samples each max*
 

 
Notes about the flow to expect after data mining:

  Select data

  Quality filtering

  Qiime 2 and OTU assignment

  Functional assignment in picrust (KEGG orthology database)
  
  Statistical analysis (mostly multivariate... I recommend you get familiar with Numerical Ecology in R)
